docs/cli-usage.md

```diff
@@ -0,0 +1,96 @@
+## Using Kubernetes from the command line
```

Comment:
I agree with Brendan (and Brian, really) that we have to have a single command line that can take a new user from having nothing to having a simple running something.  I do think there is value in an attempt to implement it without change to our existing resources (leveraging and keeping in mind the flexibility for all of these issues).  That exercise, suitably scoped to reduce future rework, seems a straight net positive.  So I would advocate to try to build the minimal 'run', review it, and iterate on to accomplish the longer term goals.  I don't think we'll get run right the first time, but we won't get it right just designing it in a vacuum :)

I'm currently working on an OpenShift equivalent of Brendan's "run" command, except instead of starting with an image, it starts with a Git repo.  In the long run, I want to be able to accomplish all of our value add / dev lifecycle additions around a solid core like run, so our wrapper tries to transform something developer focused into something operations focused.  It's been extremely helpful in highlighting feature sets that are non-obvious but core to the experience (like how image names function similarly to service names by giving a name to a particular function, and how names and namespaces overlap and can be resolved to intent).  That focus on targeting the core user (for OpenShift a web dev, for Kube a devop with an existing Docker work set) is really the key entry point to the ecosystem.

----- Original Message -----
> > @@ -0,0 +1,96 @@
> > +## Using Kubernetes from the command line
> 
> Actually, the multi-resource example above would also need a special
> transformation pass to do the magic defaulting. It mainly differs from
> generators discussed above in that it wouldn't have a specific schema, other
> than just an ordered list of related resources.
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/GoogleCloudPlatform/kubernetes/pull/3233/files#r22699936